src/celimage/dds.cpp

```diff
@@ -256,12 +256,12 @@ Image* LoadDDSImage(const fs::path& filename)
                 auto tmp = DecompressDXTc(nw, nh, format, transparent0, in);
                 if (tmp != nullptr)
                 {
-                    pixels = std::make_unique<std::uint32_t>(ddsd.width * ddsd.height);
+                    pixels = std::make_unique<std::uint32_t[]>(ddsd.width * ddsd.height);
                     // crop
                     for (std::uint32_t y = 0; y < ddsd.height; y++)
                     {
-                        std::memcpy(reinterpret_cast<char*>(pixels.get() + y * ddsd.width * 4),
-                                    reinterpret_cast<char*>(tmp.get() + y * nw * 4),
+                        std::memcpy(reinterpret_cast<char*>(pixels.get() + y * ddsd.width),
```

Comment:
The old code was

```memcpy((char *)pixels + y * ddsd.width * 4, (char *)tmp + y * nw * 4, ddsd.width * 4);```

the pointer cast was performed earlier so we don't need to * 4 in the new code.